### PR TITLE
[GCI] System Test for Login Scenario

### PR DESCRIPTION
**Pull request by MrBartusek**
This is a GCI Task https://codein.withgoogle.com/dashboard/task-instances/4569953965965312/

#### Describe the changes you have made in this pr -
I added a system test that will test if:
- user cannot log in when no credentials
- user can log in with valid credentials
- user cannot log in if one of the credentials is missing

### CODE REVIEW

File: spec/system/sign_in.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  it 'should not sing-in when no credentials' do
+    click_button 'Log in'
+
+    expect(page).to have_text('Invalid Email or password.')
+  end
+
+  it 'should sing-in when valid credentials' do
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button 'Log in'
```
[the remark this answers is not in the document]
Comment (MrBartusek) [reply spec/system/sign_in.rb]: code climate has a lot of problems with double quotes but on last issue, I was told to listen to hound and use single quote

File: spec/system/sign_in.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Sign In', type: :system do
```
[the remark this answers is not in the document]
Comment (MrBartusek) [reply spec/system/sign_in.rb]: This [has been configured](https://github.com/CircuitVerse/CircuitVerse/blob/master/.rubocop.yml#L30) to don't show so basically it doesn't matter

**Comment by MrBartusek** [comment]
@tachyons 
Done, Tests on my local machine Passed, [AppVeyor](https://ci.appveyor.com/project/ankitkataria/circuitverse/builds/29676065#L730) passed to but CircleCi failed cause it [don't have selenium configured](https://circleci.com/docs/2.0/browser-testing/#selenium)

**Comment by tachyons** [comment]
Please configure circle ci as well

File: spec/system/signin_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  before do
+    driven_by(:selenium)
+    @user = FactoryBot.create(:user)
+  end
+
+  before(:each) do
+    visit "/users/sign_in"
+  end
+
+  it "should not sing-in when no credentials" do
```
Comment (tachyons) [inline spec/system/signin_spec.rb]: sign in

**Comment by MrBartusek** [comment]
@tachyons circleci is working now

**Comment by tachyons** [comment]
Awesome :tada: just replace sing in with sign in, rest is fine

**Pull request merged**